pkg/types/kustomization.go

```diff
@@ -18,14 +18,30 @@ limitations under the License.
 package types
 
 import (
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
 	"sigs.k8s.io/kustomize/pkg/patch"
 )
 
+// TypeMeta is copied from apimachinery
```

Comment:
Full explanation e.g.
```
// TypeMeta copies k8s.io/apimachinery/pkg/apis/meta/v1.Typemeta
// Not using original to avoid bad deps - see kustomize/internal/k8sdeps/doc.go
```